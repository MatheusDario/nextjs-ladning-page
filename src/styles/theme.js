export const theme = {
  colors: {
    white: '#f0f0f2',
    blue: '#023059',
    lightblue: '#065473',
    aquablue: '#08748c',
    green: '#02735e',
    lightgreen: '#6ab04c',
    lightgray: ' #cecece',
    red: '#dc143c',
    black: '#000000',
  },
  spacing: {
    xmall: '.8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
  },
  font: {
    sizes: {
      xmall: '.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
    },
    family: {
      default: "'Lato', sans-serif",
      title: "'Montserrat', sans-serif",
    },
  },
  breakpoints: {
    mobile: '@media (max-width:768px)',
    tests: '(max-width:768px)',
  },
};
