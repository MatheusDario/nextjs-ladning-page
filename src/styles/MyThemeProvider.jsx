import { ThemeProvider } from "styled-components";
import { theme } from "./theme";
import React from "react";
import { GlobalStyles } from "./global-styles";

export default function MyThemeProvider({ children }) {
  return (
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyles />
    </ThemeProvider>
  );
}

// MyThemeProvider.propTypes = {
//   children: P.node.isRequired,
// };
