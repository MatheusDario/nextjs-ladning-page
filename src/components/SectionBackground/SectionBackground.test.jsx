import { SectionBackground } from '.';
import { renderTheme } from '../../styles/render-theme';

describe('<SectionBackground />', () => {
  it('should render with barkground dark', () => {
    const { container } = renderTheme(
      <SectionBackground $background={true}>Children</SectionBackground>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render with barkground light', () => {
    const { container } = renderTheme(
      <SectionBackground>Children</SectionBackground>,
    );
    expect(container).toMatchSnapshot();
  });
});
