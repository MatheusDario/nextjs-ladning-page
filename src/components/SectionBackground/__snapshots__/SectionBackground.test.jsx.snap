// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionBackground /> should render with barkground dark 1`] = `
.c0 {
  background: #f0f0f2;
  color: #000000;
  background: #023059;
  color: #f0f0f2;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Children
    </div>
  </div>
</div>
`;

exports[`<SectionBackground /> should render with barkground light 1`] = `
.c0 {
  background: #f0f0f2;
  color: #000000;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Children
    </div>
  </div>
</div>
`;
